kpi-parent/kpi-progress-chart/src/main/java/objective/taskboard/followup/data/FollowUpProgressDataProvider.java

```diff
@@ -36,16 +35,12 @@ public ProgressData getDataSet(String projectKey, ZoneId timezone, int projectio
 
         DashboardTimeline dashboardTimeline = timelineProvider.getDashboardTimelineForProject(projectKey, timezone);        
 
-        Optional<LocalDate> startDateOptional = dashboardTimeline.startDate;
-        if (!startDateOptional.isPresent())
-            return data;
-
-        Optional<LocalDate> endDateOptional = dashboardTimeline.endDate;
-        if (!endDateOptional.isPresent())
-            return data;
-
-        Range<LocalDate> timelineRange = RangeUtils.between(startDateOptional.get(), endDateOptional.get());
-        return filterByTimelineRange(data, timelineRange);
+        return dashboardTimeline
+                .hasBothDates((startDate, endDate) -> {
+                    Range<LocalDate> timelineRange = RangeUtils.between(startDate, endDate);
+                    return filterByTimelineRange(data, timelineRange);
+                })
+                .orElse(data);
```

Comment:
It's simpler to unterstand
```suggestion
        if (!dashboardTimeline.hasBothDates())
            return data;

        Range<LocalDate> timelineRange = RangeUtils.between(dashboardTimeline.startDate.get(), dashboardTimeline.endDate.get());
        return filterByTimelineRange(data, timelineRange);
```